internal/integration/registry_test.go

```diff
@@ -0,0 +1,169 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+//go:build integration
+
+package integration_test
+
+import (
+	"archive/tar"
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+	"sort"
+	"testing"
+
+	"github.com/google/go-containerregistry/pkg/crane"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/siderolabs/gen/xslices"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/siderolabs/image-service/pkg/configuration"
+)
+
+func testInstallerImage(ctx context.Context, t *testing.T, registry name.Registry, talosVersion, configuration string, secureboot bool, platform v1.Platform) {
+	imageName := "installer"
+	if secureboot {
+		imageName += "-secureboot"
+	}
+
+	ref := registry.Repo(imageName, configuration).Tag(talosVersion)
+
+	_, err := remote.Head(ref)
+	require.NoError(t, err)
+
+	descriptor, err := remote.Get(ref, remote.WithPlatform(platform))
+	require.NoError(t, err)
+
+	index, err := descriptor.ImageIndex()
+	require.NoError(t, err)
+
+	manifest, err := index.IndexManifest()
+	require.NoError(t, err)
+
+	platforms := xslices.Map(manifest.Manifests, func(m v1.Descriptor) string {
+		return m.Platform.String()
+	})
+
+	sort.Strings(platforms)
+
+	assert.Equal(t, []string{"linux/amd64", "linux/arm64"}, platforms)
+
+	img, err := descriptor.Image()
+	require.NoError(t, err)
+
+	layers, err := img.Layers()
+	require.NoError(t, err)
+
+	assert.Len(t, layers, 2)
+
+	assertImageContainsFiles(t, img, map[string]struct{}{
```

Comment:
should we also check for secureboot assets?

Reply:
we should, but Image Service doesn't support it yet, so we can't :) the test would fail for secureboot installer, so we'll have to fix it once we remove `t.Skip()` for secureboot installer